src/Middleware/DocBlockAnnotations.php

```diff
@@ -15,7 +15,7 @@
 
 class DocBlockAnnotations extends AbstractMiddleware
 {
-    private const DOC_BLOCK_REGEX = '/@(?P<name>[A-Za-z_-]+)(?:[ \t]+(?P<value>.*?))?[ \t]*\r?$/m';
+    private const DOC_BLOCK_REGEX = '/@(?P<name>[A-Za-z_-]+)[ \t]+(?P<value>[\w\[\]\\\\]*).*$/m';
```

Comment:
This will exclude `|null` or ` This is a description` from being set on the type

Reply:
The skipping of `|null` will break on [line 89](https://github.com/JsonMapper/JsonMapper/blob/develop/src/Middleware/DocBlockAnnotations.php#L89) The reason for the stripos is to also deal with `null|someType`

I would suggest to tweak the regex to include the `|null` (include `|` in the regex) as that is part of the annotation but leave the `This is a description` out of it
```suggestion
    private const DOC_BLOCK_REGEX = '/@(?P<name>[A-Za-z_-]+)[ \t]+(?P<value>[\w\[\]\\|]*).*$/m';
```